homeassistant/components/application_credentials/__init__.py

```diff
@@ -161,6 +160,8 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         hass, DOMAIN, _async_provide_implementation
     )
 
+    hass.config_entries.async_add_post_remove_call(async_post_remove_config_entry)
```

Comment:
Can we register this callback per config entry or per config entry domain that sets up the application credential platform so we don't need to call the callback for all config entries that are removed in the whole application?

Reply:
I've added a callback per application credentials domain, good idea.

I initially tried to do this per config entry, but did not see an easy way for application credentials to be informed. If there were a "new config entry" listener then application credentials could instead listen to that to subscribe on removals however that would invoke a callback for all config entries created (very common) in order to avoid invoking a callback when a config entry is removed (very rare)

Happy to adjust if we have better ideas, but I think your per domain suggestion is a good balance.

Reply:
@balloob what you think about this PR?